driver/pad_constant_driver.hpp

```diff
@@ -0,0 +1,389 @@
+
+    in_dev      = std::unique_ptr<GPUMem>(new GPUMem(0, input_size, sizeof(Tgpu)));
+    out_dev     = std::unique_ptr<GPUMem>(new GPUMem(0, output_size, sizeof(Tgpu)));
+    padding_dev = std::unique_ptr<GPUMem>(new GPUMem(0, 10, sizeof(size_t)));
+
+    input       = std::vector<Tgpu>(input_size, static_cast<Tgpu>(0));
+    output      = std::vector<Tgpu>(output_size, static_cast<Tgpu>(0));
+    output_host = std::vector<Tref>(output_size, static_cast<Tref>(0));
+
+    for(int i = 0; i < input.size(); i++)
+    {
+        input[i] = prng::gen_A_to_B(static_cast<Tgpu>(0.0), static_cast<Tgpu>(1.0));
+    }
+
+    if(in_dev->ToGPU(GetStream(), input.data()) != 0)
+        std::cerr << "ConstantPadDriver: Error copying data to GPU, size: " << in_dev->GetSize()
+                  << std::endl;
+
+    if(out_dev->ToGPU(GetStream(), output.data()) != 0)
+        std::cerr << "ConstantPadDriver: Error copying data to GPU, size: " << out_dev->GetSize()
+                  << std::endl;
+
+    if(padding_dev->ToGPU(GetStream(), padding.data()) != 0)
+        std::cerr << "ConstantPadDriver: Error copying data to GPU, size: "
+                  << padding_dev->GetSize() << std::endl;
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+int ConstantPadDriver<Tgpu, Tref>::RunForwardGPU()
+{
+    float kernel_total_time = 0;
+    float kernel_first_time = 0;
+
+    Timer t;
+    START_TIME
+
+    for(int i = 0; i < inflags.GetValueInt("iter"); i++)
+    {
+        miopenPadConstantFwd(GetHandle(),
+                             inputDesc,
+                             outputDesc,
+                             in_dev->GetMem(),
+                             out_dev->GetMem(),
+                             (const size_t*)padding_dev->GetMem(),
+                             value);
+
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock time elapsed: " << t.gettime_ms() / iter << " ms" << std::endl;
+
+        float kernel_avg_time = iter > 1 ? kernel_total_time / iter : kernel_first_time;
+        std::cout << "Kernel time elapsed: " << kernel_avg_time << " ms" << std::endl;
+    }
+
+    if(out_dev->FromGPU(GetStream(), output.data()) != 0)
+        std::cerr << "Error copying data from GPU, size: " << out_dev->GetSize() << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ConstantPadDriver<Tgpu, Tref>::RunForwardCPU()
+{
+    mloConstantPadForwardRunHost(
+        inputDesc, outputDesc, input.data(), output_host.data(), padding.data(), value);
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ConstantPadDriver<Tgpu, Tref>::VerifyForward()
```

Comment:
Try using miopen::rms_range. Don't be afraid to reference other people's code. The argmax driver has an example of checking if values are equal.

Reply:
I’ve switched to rms_range for the comparision. Please check the updated code